### PR TITLE
Add user agent:

### PR DESCRIPTION
**Pull request by outscale-vbr**
* add user agent
* add version
* add version test

**What type of PR is this?**
/kind feature   

**What this PR does / why we need it**:
Add user agent to catch hen cluster api is used.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Review by outscale-hmi — COMMENTED** [review]
OK

File: Makefile [hunk 1]
```diff
@@ -48,7 +48,10 @@ GOBIN=$(shell go env GOPATH)/bin
 else
 GOBIN=$(shell go env GOBIN)
 endif
-
+VERSION := ${IMAGE_TAG}
+GIT_COMMIT ?= $(shell git rev-parse HEAD)
+BUILD_DATE ?= $(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
+LDFLAGS ?= "-X ${PKG}/pkg/util.capoVersion=${RELEASE_TAG} -X ${PKG}/pkg/util.gitCommit=${GIT_COMMIT} -X ${PKG}/pkg/util.buildDate=${BUILD_DATE}"
```
Comment (outscale-mdr) [inline Makefile:54]: Where do you initialized  "PKG" ?

File: cloud/scope/clients.go [hunk 2]
```diff
@@ -2,10 +2,11 @@ package scope
 
 import (
 	"context"
-	"os"
-
 	"errors"
+	"fmt"
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/util/reconciler"
```
Comment (outscale-mdr) [inline cloud/scope/clients.go:7]: ".git" ? Oh it is in your module name

File: util/reconciler/version_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	version := GetVersion()
+
+	expected := VersionInfo{
+		CapoVersion: "",
+		GitCommit:   "",
+		BuildDate:   "",
+		GoVersion:   runtime.Version(),
+		Compiler:    runtime.Compiler,
+		Platform:    fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+	}
```
Comment (outscale-mdr) [inline util/reconciler/version_test.go:21]: I don't see the use of this test, why do you want to test this feature ?

File: Makefile (hunk 1 above)
Comment (outscale-vbr) [reply Makefile:55]: Done

File: Makefile [hunk 4]
```diff
@@ -48,7 +48,11 @@ GOBIN=$(shell go env GOPATH)/bin
 else
 GOBIN=$(shell go env GOBIN)
 endif
-
+PKG := github.com/outscale-dev/cluster-api-provider-outscale
```
Comment (outscale-mdr) [inline Makefile:51]: I think there is a ".git" at the end

Comment (outscale-mdr) [reply Makefile:51]: Can you confim that the user-agent is changed, you can see that in the logs

File: util/reconciler/version_test.go (hunk 3 above)
Comment (outscale-mdr) [reply util/reconciler/version_test.go:21]: What do you think, is it useful ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

File: cloud/scope/clients.go [hunk 5]
```diff
@@ -24,8 +24,15 @@ func newOscClient() (*OscClient, error) {
...
+	if version == "" {
+		return nil, errors.New("Version is required")
+	}
+	config := osc.NewConfiguration()
+	config.Debug = true
+	config.UserAgent = fmt.Sprintf("cluster-api-provider-outscale/%s", version)
 	oscClient := &OscClient{
-		api: osc.NewAPIClient(osc.NewConfiguration()),
+		api: osc.NewAPIClient(config),
 		auth: context.WithValue(context.Background(), osc.ContextAWSv4, osc.AWSv4{
```
Comment (outscale-hmi) [inline cloud/scope/clients.go:36]: we have a helper to get creds from env variables or from json file: 
NewConfigEnv() or NewConfigFile , and then methods for configuration() and context() to get creds properly.

**Review by outscale-hmi — COMMENTED** [review]
OK

**Review by outscale-hmi — APPROVED** [review]
OK

**Comment by outscale-vbr** [comment]
Closes https://github.com/outscale-dev/cluster-api-provider-outscale/issues/2

**Pull request merged**